src/Controller/OcrController.php

```diff
@@ -36,13 +38,19 @@ class OcrController extends AbstractController
      * @param string $endpoint
      * @param string $key
      */
-    public function __construct(RequestStack $requestStack, Intuition $intuition, string $endpoint, string $key)
-    {
-        $request = $requestStack->getCurrentRequest();
-
+    public function __construct(
+        RequestStack $requestStack,
+        Intuition $intuition,
+        EngineFactory $engineFactory
+    ) {
         // Dependencies.
         $this->intuition = $intuition;
-        $this->engine = new GoogleCloudVisionEngine($endpoint, $key);
+
+        $request = $requestStack->getCurrentRequest();
+
+        // Engine.
+        $this->engine = $engineFactory->get($request->get('engine'));
```

Comment:
Should we set a default here, ala `$request->get('engine', 'google')`? Otherwise I get `Argument 1 passed to App\Engine\EngineFactory::get() must be of the type string, null given` when browsing to the home path

Reply:
Oops! Yes, good idea. Done.